src/Utils/getEdgeIntegralOfMagneticDipole.jl

```diff
@@ -0,0 +1,83 @@
+export getEdgeIntegralOfMagneticDipole
+
+"""
+        function MaxwellFrequency.Utils.getEdgeIntegralOfMagneticDipole
+
+        s = getEdgeIntegralOfMagneticDipole(mesh,dipoleLoc,dipoleMoment)
+        s = getEdgeIntegralOfMagneticDipole(mesh,dipoleLoc,dipoleMoment,normalize)
+
+        Compute the integral of a piecewise linear edge grid basis projection
+        for the modeling of a magnetic dipole. This function can be used to evaluate the
+        source and receiver term of an infinitly small closed loop in electromagnetic
+        modelling.
+
+        This function is built on getEdgeIntegralOfPolygonalChain. The dipole 
+        is first located in one cell of the mesh, then the 12 edges of the cell
+        are used to model 6 orthogonal loops on P faces, with 2 loops for each 
+        direction. The x-component moment is distributed to the two x-facing 
+        loops according to the relative x-location of the dipole within the cell.
+
+        INPUT
+        mesh: tensor mesh or octree mesh
+        dipoleLoc: a (x,y,z) vector of the location of the dipole
+        dipoleMoment: a (x,y,z) vector of the vector moment of the dipole
+        normalize: if true, model a unit dipole by regardless of the norm of dipolo moment (boolean, optional)
+
+        OUTPUT
+        s: source vector
+
+        NOTE (Dikun)
+        Currently only work with octree mesh, because tensor mesh dones't have 'getNodeToCellCenteredMatrix'.
+
+"""
+function getEdgeIntegralOfMagneticDipole(mesh::OcTreeMesh, dipoleLoc::Array{Float64,1},dipoleMoment::Array{Float64,1}; normalize=true)
+
+    if normalize
+        moment = dipoleMoment / norm(dipoleMoment)
+    else
+        moment = dipoleMoment
+    end
+
+    cc = getCellCenteredGrid(mesh)
+    distance = (cc[:,1]-dipoleLoc[1]).^2 + (cc[:,2]-dipoleLoc[2]).^2 + (cc[:,3]-dipoleLoc[3]).^2 
+    ccInd = sortperm(distance)[1] # cell center index of the cube that encloses the loc
+    nodeInd = find(getNodeToCellCenteredMatrix(mesh)[ccInd,:]) # find the neighboring nodes's indices
```

Comment:
Both JOcTree and jInv.Mesh have a function called `getNodalAverageMatrix` that take a mesh as input. In JOcTree, `getNodalAverageMatrix` is just a wrapper around `getNodeToCellCenteredMatrix` that checks to see whether the matrix needs to be computed or not. So, assuming there aren't any other incompatibilities, changing this line to
```
nodeInd = find(getNodalAverageMatrix(mesh)[ccInd,:])
```
should make this mesh independent.